addons/stock_picking_batch/models/stock_picking_batch.py

```diff
@@ -118,3 +117,13 @@ class StockPicking(models.Model):
         'stock.picking.batch', string='Batch Picking', oldname="wave_id",
         states={'done': [('readonly', True)], 'cancel': [('readonly', True)]},
         help='Batch associated to this picking', copy=False)
+
+    def action_done(self):
```

Comment:
I don't think this is necessary in v11, and looks more an IMP than a FIX (master or v13 would be ok). In the original commit, the behaviour was already known: 0f9f3ded264a0587fef74bfba361edcf71dad585

By the way, if you want to go in this direction, why not a computed field? Something like:

```python
@api.depends('picking_ids.state')
def _compute_state(self):
    for batch in self:
        batch.state = bool(batch.picking_ids and all(p.state in ['cancel', 'done'] for p in batch.picking_ids))
```

Reply:
@nim-odoo we did all that in master in https://github.com/odoo/odoo/pull/42582 fyi